src/DataStore/DefaultDataStore.php

```diff
@@ -226,34 +221,50 @@ protected function qualify($href)
 
     private function executeRequest($httpMethod, $href, $body = '', array $query = array())
     {
-        $request = new DefaultRequest(
-                       $this->apiKey,
-                       $httpMethod,
-                       $href,
-                       $query,
-                       array(),
-                       $body,
-                       strlen($body));
+        $headers = [];
+        $headers['Accept'] = 'application/json';
+
+        $userAgentBuilder = new UserAgentBuilder;
+        $headers['User-Agent'] = $userAgentBuilder->setOsName(php_uname('s'))
+            ->setOsVersion(php_uname('r'))
+            ->setPhpVersion(phpversion())
+            ->build();
+
+        if ($body) {
+            $headers['Content-Type'] = 'application/json';
+
+            if (strpos($href, '/oauth/token')) {
+                $arr = json_decode($body);
+                $arr = (array) $arr;
+                ksort($arr);
+                $body = http_build_query($arr);
+
+                $headers['Content-Type'] = 'application/x-www-form-urlencoded';
+            }
+        }
 
-        $this->applyDefaultRequestHeaders($request);
+        $uri = new Uri($href);
+        foreach ($query as $key => $value) {
+            $uri = Uri::withQueryValue($uri, $key, $value);
+        }
+
+        $request = new Request($httpMethod, $uri, $headers, $body);
```

Comment:
Use the request factory instead. 

`MessageFactoryDiscovery::find()->createRequest()`